### PR TITLE
add configuration diagrams to clusterapi readme

### PR DESCRIPTION
**Pull request by elmiko**
This change adds ascii diagrams to help illustrate the differences
between the various authentication configurations for the clusterapi
provider. Due to the distributed nature of Cluster API and its ability
to have several Kubernetes clusters managed from a central location, the
kubeconfig authentication options for it are slightly more complex than
other providers.

### CODE REVIEW

**Comment by elmiko** [comment]
@jackfrancis this is something we had talked about quite a long time ago, i got inspired ;)

File: cluster-autoscaler/cloudprovider/clusterapi/README.md [hunk 1]
```diff
@@ -66,30 +66,80 @@ To specify the kubeconfig path for the management cluster to monitor, use the
...
 ```
-cluster-autoscaler --cloud-provider=clusterapi --kubeconfig=/mnt/workload.kubeconfig --cloud-config=/mnt/management.kubeconfig
+cluster-autoscaler --cloud-provider=clusterapi \
+                   --kubeconfig=/mnt/workload.kubeconfig \
+                   --cloud-config=/mnt/management.kubeconfig
+```
+
+### Autoscaler running anywhere, with a common kubeconfig for management and workload clusters
+```
++--------+             +------------+             +----------+
```
Comment (enxebre) [inline cluster-autoscaler/cloudprovider/clusterapi/README.md:131]: if they share the same kubeconfig I'd expect this to look like the box in https://github.com/kubernetes/autoscaler/pull/4474/files#diff-eabe434726684600ca7ec154b4f511caa551401fd78268f48a70a812f2ec8040R69 for mgmt / workload and have another box for the autoscaler.

**Comment by enxebre** [comment]
Nice, thanks!
/area provider/cluster-api

File: cluster-autoscaler/cloudprovider/clusterapi/README.md (hunk 1 above)
Comment (elmiko) [reply cluster-autoscaler/cloudprovider/clusterapi/README.md:131]: yeah, i went back and forth about the designs, i wanted to illustrate that they don't necessarily have to be the same, but i could make it more like the diagram above if you think that makes more sense?

**Comment by elmiko** [comment]
update
* changed common kubeconfig diagram to show a single source for mgmt/workload cluster

**Comment by shysank** [comment]
/lgtm
Thanks for doing this!

**Pull request merged**